cpp/tests/io/parquet_test.cpp

```diff
@@ -4215,4 +4280,194 @@ TEST_F(ParquetReaderTest, EmptyColumnsParam)
+  cols.push_back(std::make_unique<cudf::column>(static_cast<cudf::column_view>(list_int_col)));
+  cols.push_back(std::make_unique<cudf::column>(static_cast<cudf::column_view>(list_int_col)));
+  auto expected_mixed = std::make_unique<table>(std::move(cols));
+  EXPECT_EQ(5, expected_mixed->num_columns());
+
+  cudf_io::parquet_reader_options in_opts =
+    cudf_io::parquet_reader_options::builder(cudf_io::source_info{filepath})
+      .convert_binary_to_strings({true, true, true, true, true});
+  auto result = cudf_io::read_parquet(in_opts);
+
+  CUDF_TEST_EXPECT_TABLES_EQUAL(expected_string->view(), result.tbl->view());
+
+  cudf_io::parquet_reader_options default_in_opts =
+    cudf_io::parquet_reader_options::builder(cudf_io::source_info{filepath});
+  result = cudf_io::read_parquet(default_in_opts);
+
+  CUDF_TEST_EXPECT_TABLES_EQUAL(expected_string->view(), result.tbl->view());
+
+  cudf_io::parquet_reader_options mixed_in_opts =
+    cudf_io::parquet_reader_options::builder(cudf_io::source_info{filepath})
+      .convert_binary_to_strings({true, true, true, false, false});
+  result = cudf_io::read_parquet(mixed_in_opts);
+
+  CUDF_TEST_EXPECT_TABLES_EQUAL(expected_mixed->view(), result.tbl->view());
+}
+
+TEST_F(ParquetReaderTest, NestedByteArray)
+{
+  constexpr auto num_rows = 8;
+
+  auto seq_col0 = random_values<int>(num_rows);
+  auto seq_col2 = random_values<float>(num_rows);
+  auto seq_col3 = random_values<int8_t>(num_rows);
+  auto validity = cudf::detail::make_counting_transform_iterator(0, [](auto i) { return true; });
+
+  column_wrapper<int> int_col{seq_col0.begin(), seq_col0.end(), validity};
+  column_wrapper<float> float_col{seq_col2.begin(), seq_col2.end(), validity};
+  cudf::test::lists_column_wrapper<int8_t> list_list_int_col{
+    {{'M', 'o', 'n', 'd', 'a', 'y'},
+     {'W', 'e', 'd', 'n', 'e', 's', 'd', 'a', 'y'},
+     {'F', 'r', 'i', 'd', 'a', 'y'}},
+    {{'M', 'o', 'n', 'd', 'a', 'y'}, {'F', 'r', 'i', 'd', 'a', 'y'}},
+    {{'M', 'o', 'n', 'd', 'a', 'y'},
+     {'W', 'e', 'd', 'n', 'e', 's', 'd', 'a', 'y'},
+     {'F', 'r', 'i', 'd', 'a', 'y'}},
+    {{'F', 'r', 'i', 'd', 'a', 'y'},
+     {'F', 'r', 'i', 'd', 'a', 'y'},
+     {'F', 'u', 'n', 'd', 'a', 'y'}},
+    {{'M', 'o', 'n', 'd', 'a', 'y'},
+     {'W', 'e', 'd', 'n', 'e', 's', 'd', 'a', 'y'},
+     {'F', 'r', 'i', 'd', 'a', 'y'}},
+    {{'F', 'r', 'i', 'd', 'a', 'y'},
+     {'F', 'r', 'i', 'd', 'a', 'y'},
+     {'F', 'u', 'n', 'd', 'a', 'y'}},
+    {{'M', 'o', 'n', 'd', 'a', 'y'},
+     {'W', 'e', 'd', 'n', 'e', 's', 'd', 'a', 'y'},
+     {'F', 'r', 'i', 'd', 'a', 'y'}},
+    {{'M', 'o', 'n', 'd', 'a', 'y'}, {'F', 'r', 'i', 'd', 'a', 'y'}}};
+
+  std::vector<std::unique_ptr<column>> cols;
+  cols.push_back(int_col.release());
+  cols.push_back(float_col.release());
+  cols.push_back(list_list_int_col.release());
+  auto expected = std::make_unique<table>(std::move(cols));
```

Comment:
This can be shortened, since we only need a `table_view`, for the purpose of the test:
```suggestion
  auto const expected = table_view{{int_col, float_col, list_list_int_col}};
```

Reply:
We'll need to change things downstream to use `expected` as a `table_view` directly, instead of a `std::unique_ptr<table>`.